internal/app/controller.go

```diff
@@ -0,0 +1,37 @@
+package app
+
+import (
+	"github.com/asynkron/protoactor-go/actor"
+
+	"github.com/ccamel/playground-protoactor.go/internal/middleware"
+)
+
+// Controller is the actor responsible for managing the lifecycle of the applications.
+type Controller struct {
+	middleware.SpawnAwareMixin
+}
+
+func (state *Controller) Receive(context actor.Context) {
+	switch context.Message().(type) {
+	case *actor.Started:
+		for k, v := range All() {
+			state.SpawnNamedOrDie(context, actor.PropsFromProducer(v(nil)), k)
+		}
+	case *actor.Stopping:
+	case *actor.Stopped:
+	case *actor.Restarting:
+	}
+}
+
+func Props() *actor.Props {
+	supervisor := actor.NewOneForOneStrategy(5, 1000, func(_ interface{}) actor.Directive {
+		return actor.RestartDirective
+	})
+
+	return actor.
+		PropsFromProducer(
+			func() actor.Actor {
+				return &Controller{}
+			},
+			actor.WithSupervisor(supervisor))
+}
```

Comment:
Oh, yes, I’m using magic (and quite arbitrary) constants because, let’s say, this is just a playground, so I haven’t pushed the approach further. Ideally, these values should be configurable through a settings file, for instance, which I plan to address in a future update.

Additionally, there’s no need for extra logging here, as the lifecycle and certain system decisions (like supervision) are already logged elsewhere. I’d rather not add unnecessary "noise".